### ISSUE TITLE
MySQL 5.7: Failing tests with Cannot get geometry object from data you send to the GEOMETRY field

### ISSUE THREAD

**Issue by schultyy** [sozialhelden/wheelmap#355]
When running the tests with MySQL 5.7, we encounter the following error:

```
Mysql2::Error: Cannot get geometry object from data you send to the GEOMETRY field: INSERT INTO `regions` (`created_at`, `grenze`, `lft`, `name`, `rgt`, `slug`, `updated_at`) VALUES ('2016-08-08 14:04:51', GeomFromWKB(0x00000000030000000100000005402a2d36f7e3d1cc404a2b463497b741402b8595feda6613404a2b463497b741402b8595feda6613404a567620ee8d11402a2d36f7e3d1cc404a567620ee8d11402a2d36f7e3d1cc404a2b463497b741,0), 1, 'Berlin', 2, 'berlin', '2016-08-08 14:04:51')
```

**Comment by Xylakant** [comment]
Can we identify the text representation of this polygon that we're trying to insert here?

**Comment by Hoverbear** [comment]
Possibly relevant MySQL bugs:

* https://bugs.mysql.com/bug.php?id=79060
* http://stackoverflow.com/questions/34311195/mysql-import-cannot-get-geometry-object-from-data-you-send-to-the-geometry-fiel

The bug listed appears to have been closed however it doesn't seem like it was fixed.

**Comment by Xylakant** [comment]
So the fundamental problem is that the data we're providing is a polygon which matches what we're trying to actually insert https://github.com/sozialhelden/wheelmap/blob/master/spec/factories/region_factory.rb#L5:
         
    mysql> SELECT ST_AsText(GeomFromWKB(0x00000000030000000100000005402a2d36f7e3d1cc404a2b463497b741402b8595feda6613404a2b463497b741402b8595feda6613404a567620ee8d11402a2d36f7e3d1cc404a567620ee8d11402a2d36f7e3d1cc404a2b463497b741,0)) AS shape;
    +------------------------------------------------------------------------------------------------------+
    | shape                                                                                                |
    +------------------------------------------------------------------------------------------------------+
    | POLYGON((13.08831 52.33808,13.76091 52.33808,13.76091 52.67548,13.08831 52.67548,13.08831 52.33808)) |
    +------------------------------------------------------------------------------------------------------+
    1 row in set, 1 warning (0.00 sec)

and we're trying to insert that into a multipolygon column - this seems to be valid in mysql up to 5.6 but not in 5.7. Using a polygon works:

    CREATE TABLE `regions2` (
      `id` bigint(20) NOT NULL AUTO_INCREMENT,
      `name` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL,
      `grenze` polygon NOT NULL,
      `created_at` datetime DEFAULT NULL,
      `updated_at` datetime DEFAULT NULL,
      `parent_id` bigint(20) DEFAULT NULL,
      `admin_level` int(11) DEFAULT NULL,
      `lft` int(11) DEFAULT NULL,
      `rgt` int(11) DEFAULT NULL,
      `depth` int(11) DEFAULT NULL,
      `slug` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL,
      PRIMARY KEY (`id`),
      UNIQUE KEY `index_regions_on_slug` (`slug`)
    ) ENGINE=InnoDB AUTO_INCREMENT=266 DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;


    mysql> INSERT INTO `regions2` (`created_at`, `grenze`, `lft`, `name`, `rgt`, `slug`, `updated_at`) VALUES ('2016-08-08 14:04:51', GeomFromWKB(0x00000000030000000100000005402a2d36f7e3d1cc404a2b463497b741402b8595feda6613404a2b463497b741402b8595feda6613404a567620ee8d11402a2d36f7e3d1cc404a567620ee8d11402a2d36f7e3d1cc404a2b463497b741,0), 1, 'Berlin', 2, 'berlin', '2016-08-08 14:04:51');
    Query OK, 1 row affected, 1 warning (0.00 sec)

**Comment by Xylakant** [comment]
in the existing data we do have a mixture of polygon/multipolygon shapes. this makes things harder :(

**Comment by Hoverbear** [comment]
It's reasonable to assume they should all be `multipolygon` if we do migrate? A single polygon is still a multipolygon, but a multipolygon is not always a single polygon.

**Comment by Xylakant** [comment]
Probably migrating to geometry might be the better option if that's supported by the adapter and if we have a proper migration path there (for example it would need to be supported on some version that we can install so we can migrate the existing data in a controlled fashion)

**Comment by Hoverbear** [comment]
I agree, there's definitely a more complicated migration needed once we know which way to go.
